.changeset/sharp-beds-lie.md

```diff
@@ -0,0 +1,23 @@
+---
+"@meilisearch/instant-meilisearch": minor
+---
+
+- exposed created `MeiliSearch` instance
+- re-exported all of `"meilisearch"` package
+
+These improvements make it so that no separate `"meilisearch"` package
+has to be installed in order to use its exports directly. This
+way a single `MeiliSearch` instance can be re-used, and we can potentially
+save on bundle size by avoiding a separate different version installation of
+`"meilisearch"`.
+
+```typescript
+import { instantMeiliSearch, meilisearch } from '@meilisearch/instant-meilisearch'
```

Comment:
Is my suggestion correct right?! 
```suggestion
import { instantMeiliSearch, meiliSearchInstance } from '@meilisearch/instant-meilisearch'
```

Reply:
The `meilisearch` from `import { instantMeiliSearch, meilisearch } from '@meilisearch/instant-meilisearch'` is the object that contains all of the re-exported elements from `"meilisearch"` package as seen in `index.ts`:
```typescript
// ...
export * as meilisearch from 'meilisearch'
```

The instance is what is created by the main function `instantMeiliSearch()`:

```typescript
instantMeiliSearch(/*...*/).meiliSearchInstance
//                                      ^ this thing, the instance which is also used by the `searchClient`,
//                                          which is then passed to `instantsearch`
```

I could have re-exported it as is (`export * from 'meilisearch'`), but I wanted to avoid any export conflicts, so I made a named re-export `export * as meilisearch from 'meilisearch'`